### PR TITLE
WD-16: Move to DynamoDB!

### PR DESCRIPTION
**Pull request by cakoshakib**
Main contribution is `dynamo.go` which interfaces with dynamodb and a change in the auth flow

### CODE REVIEW

File: api/tasks.go [hunk 1]
```diff
@@ -58,27 +78,36 @@ func UpdateTask(c *gin.Context) {
...
-	task := db.SelectUserTaskByID(uid, taskid)
-	if !db.DeleteTask(uid, taskid) {
-		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "task with id " + taskid + " not found"})
+	task, err := db.GetUserTaskByID(uid, taskid)
+	if err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "error getting task to delete " + taskid})
+		return
+	}
+	if err := db.DeleteTask(uid, taskid); err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "failed to delete " + taskid + " in db (maybe wrong id?)"})
```
Comment (markdstouffer) [inline api/tasks.go:110]: do enums exist in go? if so, export an enum with all error messages. _We don't like magic strings_

File: handler/router.go [hunk 2]
```diff
@@ -22,7 +22,9 @@ func bind_discord(router *gin.Engine) {
 
 func set_cors(router *gin.Engine) {
 	config := cors.DefaultConfig()
-	config.AllowOrigins = []string{"http://localhost:5173", "http://localhost:8080"}
+	config.AllowOrigins = []string{"http://localhost:5173", "http://localhost:8080", "http://127.0.0.1:5173", "https://068a6tnn1g.execute-api.us-east-2.amazonaws.com/staging"}
+	config.ExposeHeaders = []string{"set-cookie"}
```
Comment (markdstouffer) [inline handler/router.go:26]: 😭

**Review by markdstouffer — APPROVED** [review]
LGTM!

File: api/tasks.go (hunk 1 above)
Comment (cakoshakib) [reply api/tasks.go:110]: will do in logging ticket 🫡

**Pull request merged**